### PR TITLE
Some Relay-modernizing [#6211]

### PR DESCRIPTION
**Pull request by adamhooper**
Modeling after my `<MoveProjectMediaToProjectAction>`, here are more Relay-modernizations:

* `<AddProjectMediaToProjectAction>`
* `<RemoveProjectMediaFromProjectAction>`
* `<EditTitleAndDescriptionDialog>`
* Rename `<MediaActions>` to `<MediaActionsMenuButton>`. Make it a fragment container, and move `<EditTitleAndDescriptionDialog>` to be its child. (TODO make _all_ its actions fragment-container children.)
* Fix a forwardRef JS warning from `withSetFlashMessage()` [#8179]
* Fix a couple of propTypes [#8179]

### CODE REVIEW

File: src/app/components/media/EditTitleAndDescriptionDialog.js [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+      onFailure: (errors) => {
+        console.error(errors); // eslint-disable-line no-console
+        setFlashMessage((
+          <FormattedMessage
+            id="mediaDetail.editReportError"
+            defaultMessage="Sorry, an error occurred while updating the item. Please try again and contact {supportEmail} if the condition persists."
+            values={{ supportEmail: stringHelper('SUPPORT_EMAIL') }}
+          />
+        ));
+      },
```
Comment (amoedoamorim) [inline src/app/components/media/EditTitleAndDescriptionDialog.js:131]: Our typical handling is to extract a message from the request response and use unknownError or a more context-specific message as a fallback.

File: src/app/components/media/AddProjectMediaToProjectAction.js [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        setIsSaving(false);
+        console.error(errors); // eslint-disable-line no-console
+        setFlashMessage((
+          <FormattedGlobalMessage
+            messageKey="unknownError"
+            values={{ supportEmail: stringHelper('SUPPORT_EMAIL') }}
+          />
+        ));
+      },
+    });
```
[the remark this answers is not in the document]
Comment (adamhooper) [reply src/app/components/media/AddProjectMediaToProjectAction.js:107]: Hear, hear! I'll build a helper method in `FlashMessage.js`.

Hopefully we can get a new pattern someday ... but I think the way to do that is one mutation API at a time. For now the helper will be fantastic.

[the remark this answers is not in the document]
Comment (adamhooper) [reply src/app/components/media/AddProjectMediaToProjectAction.js]: I put it in `helpers.js`, alongside the Relay Classic one.

Trying a new calling convention:

```
setFlashMessage(getErrorMessageFromRelayModernProblem(errors) || (
  <FormattedGlobalMessage
    messageKey="unknownError"
    values={{ supportEmail: stringHelper('SUPPORT_EMAIL') }}
  />
));
```

What do you think?

[the remark this answers is not in the document]
Comment (amoedoamorim) [reply src/app/components/media/AddProjectMediaToProjectAction.js]: I think we can do a bit better, imo. We shouldn't need to pass `unknownError` message every time we call the error handling/signaling, the helper should do it. We'd only pass something if we wanted to override the `unknownError` with something more specific.

**Pull request merged**